README.md

```diff
@@ -73,8 +73,77 @@ git checkout HEAD examples/simple/pi.cc
 
 ### Applying Dredd to a CMake project
 
-TODO(https://github.com/mc-imperial/dredd/issues/28): add instructions
-on how to apply dread to a larger project
+The `examples/math` directory contains a small math library to illustrate how to apply Dredd to a larger project.
+To compile the math library, run the following from the root of the repository:
+```
+cd examples/math
+mkdir build && cd build
+cmake -G Ninja -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
+cmake --build .
+```
+The `-DCMAKE_EXPORT_COMPILE_COMMANDS=ON` argument causes CMake to output a compilation database called `compile_comands.json` to be output to the `build` directory.
+You can read more about compilation databases for various build systems [here](https://sarcasm.github.io/notes/dev/compilation-database.html).
+
+You can check the output of the tests before mutations have been applied by running 
+`./mathtest/mathtest` from the build directory.
+
+To apply mutants to a single file in the library, run the following command:
+```
+path/to/dredd/repo/third_party/clang+llvm-13.0.1/bin/dredd -p compile_commands.js <path-to-file>
+``` 
+For example, `../../../third_party/clang+llvm-13.0.1/bin/dredd ../math/src/exp.cc` will apply
+mutants to the file `exp.cc`.
+
+To view the changes that Dredd has made you can do `git status` to see that `exp.cc` has changed, and `git diff` to see
+the effect that Dredd has had on this file. These changes will be hard to understand as they are not intended to be
+readable by humans.
+
+The `-p` option allows the compilation database generated by CMake above to be passed to Dredd.
+This is so that Dredd knows the correct compiler options to use when processing each source file.
+
+If you wish to mutate multiple source files simultaneously,pass them all to a single invocation of Dredd. 
+For the example project we provide a shell script to do this, `mutate.sh`.
+
+To revert the changes that Dredd made to `exp.cc`, run:
+```
+git checkout HEAD ../math/src/exp.cc
+git status ../math/src/exp.cc
```

Comment:
Add a comment before this command, like:

```
# Confirm that git now regards the file as having no modifications
```

Reply:
Done